### ISSUE TITLE
Capture in IOS stopped working

### ISSUE THREAD

**Issue by gorghoa** [cordova-plugin-camera-preview/cordova-plugin-camera-preview#278]
Hi,
Sorry to bother you on a (maybe) sunny friday afternoon…

I don’t know why it suddenly stopped working (was working fine). I’ve uprgaded to latest xcode version recently, the testing device is an iphone 5S running latest ios version, my plugin checkout is up to date with master.

The preview starts well, but when I programatically call `takeCapture`, I have this error in xdebug’s console:

```
2017-04-21 15:08:32.538514+0200 MyApp[809:128542] THREAD WARNING: ['CameraPreview'] took '99.180176' ms. Plugin should use a background thread.
2017-04-21 15:08:32.539044+0200 MyApp[809:128776] defaultCamera: rear

2017-04-21 15:08:33.222444+0200 MyApp[809:128542] takePicture
2017-04-21 15:08:33.360513+0200 MyApp[809:128542] Done creating still image
2017-04-21 15:08:33.360974+0200 MyApp[809:128542] Error Domain=AVFoundationErrorDomain Code=-11800 "The operation could not be completed" UserInfo={NSUnderlyingError=0x174446a80 {Error Domain=NSOSStatusErrorDomain Code=-16800 "(null)"}, NSLocalizedFailureReason=An unknown error occurred (-16800), NSLocalizedDescription=The operation could not be completed}
```

The callbacks are never called (nor the success, nor the error)

I have absolutely no idea what is going wrong…

thanks :)

**Comment by westonganger** [comment]
Did you manage to fix it?

**Comment by sertal70** [comment]
Hi,
I have a similar issue, in my case the error message is a little bit different:

```
2017-05-12 15:33:57.350 CameraApp[33947:2758410] Error Domain=AVFoundationErrorDomain Code=-11814 "Cannot Record" UserInfo={NSLocalizedDescription=Cannot Record, NSLocalizedRecoverySuggestion=Try recording again.}

```
this error happens only on the simulator, crashing the app. If I attach my iPad for debug everything goes fine (picture is taken and the app continue to work well).

I tried to debug the code in CameraPreview but I got lost in an assembler code line after that the takePicture routine has been executed (apparently without errors).

It's not a blocking issue because I can debug my app directly on device, but is very annoying ... also it could potentially lead to a bigger issue elsewhere, who knows?

**Comment by westonganger** [comment]
@sertal70 Please read bottom of Readme. On ios Sim you are not allowed to use camera, you must use ios device. Apple doesn't allow it unless something has changed recently.

**Comment by sertal70** [comment]
@westonganger thanks, I didn't notice the warning in the readme. 

The thing is that I started to use the plugin last January and I were able to debug my app using the simulator without issues, until two weeks ago when I upgraded my project to Ionic3 (upgrading also all plugin used in the project including camera-preview).

Maybe something has been changed in the meantime between January and April in the plugin?

**Comment by westonganger** [comment]
I dont think so. It has been this way for as long as I can remember. Maybe you were using Android simulator.

**Comment by sertal70** [comment]
@westonganger, I resumed from backup the project before ionic 3 migration, started the debug with simulator (iPhone SE, not Android) and the issue is not present.
